nym-vpn/ui/src/pages/location/NodeLocation.tsx

```diff
@@ -0,0 +1,105 @@
+import { useTranslation } from 'react-i18next';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { invoke } from '@tauri-apps/api';
+import { useMainDispatch, useMainState } from '../../contexts';
+import { InputEvent, NodeHop } from '../../types/general';
+import { Country, StateDispatch } from '../../types';
+import { routes } from '../../constants';
+import SearchBox from './SearchBox';
+import CountryList from './CountryList';
+import QuickConnect from './QuickConnect';
+
+function NodeLocation({ type }: NodeHop) {
+  const isEntryNodeSelectionScreen = type === 'entry';
+  const { t } = useTranslation('nodeLocation');
+  const [countries, setCountries] = useState<Country[]>([]);
+  const [search, setSearch] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [foundCountries, setFoundCountries] = useState<Country[]>([]);
+
+  const state = useMainState();
+  const dispatch = useMainDispatch() as StateDispatch;
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    setLoading(true);
+    const getNodeCountries = async () => {
+      const countries = await invoke<Country[]>('get_node_countries');
+      setTimeout(() => {
+        setCountries(countries);
+        setFoundCountries(countries);
+        setLoading(false);
+      }, 1000);
+    };
+    getNodeCountries().catch(console.error);
+  }, []);
+
+  const filter = (e: InputEvent) => {
+    const keyword = e.target.value;
+    if (keyword !== '') {
+      const results = countries.filter((country) => {
+        return country.name.toLowerCase().startsWith(keyword.toLowerCase());
+        // Use the toLowerCase() method to make it case-insensitive
+      });
+      setFoundCountries(results);
+    } else {
+      setFoundCountries(countries);
+      // If the text field is empty, show all users
+    }
+    setSearch(keyword);
+  };
+
+  const isCountrySelected = (code: string): boolean => {
+    return isEntryNodeSelectionScreen
+      ? isCountrySelectedEntryNode(code)
+      : isCountrySelectedExitNode(code);
+  };
+
+  const isCountrySelectedEntryNode = (code: string): boolean => {
+    return state.localAppData.entryNode?.id === code;
+  };
+
+  const isCountrySelectedExitNode = (code: string): boolean => {
+    return state.localAppData.exitNode?.id === code;
+  };
```

Comment:
suggestion: when there is no concrete reason to extract code into dedicated function, avoid doing it, it's probably an antipattern, like those one-liners/one-checks above

If your initial motivation was to avoid having to rewrite that kinda long checks
use property "destructuring" instead, to bring into scope what you need

```
  const {
    localAppData: { entryNode, exitNode },
  } = useMainState();

// and so
entryNode?.id === code
exitNode?.id === code
```